app/src/main/java/com/example/tasky/feature_authentication/domain/use_case/Login.kt

```diff
@@ -0,0 +1,36 @@
+package com.example.tasky.feature_authentication.domain.use_case
+
+import com.example.tasky.feature_authentication.data.util.AuthResult
+import com.example.tasky.feature_authentication.domain.repository.AuthRepository
+import com.example.tasky.feature_authentication.domain.util.UseCaseResult
+import com.example.tasky.feature_authentication.domain.validation.UserDataValidator
+
+class Login(
+    private val repository: AuthRepository,
+    private val userDataValidator: UserDataValidator
+) {
+
+    suspend operator fun invoke(email: String, password: String): UseCaseResult {
+
+        val emailChecker = userDataValidator.validateEmail(email)
+        val passwordChecker = userDataValidator.validatePassword(password)
+
+        if(!emailChecker.isValid) {
+            return UseCaseResult.ErrorEmail(emailChecker.emailError
+                ?: return UseCaseResult.GenericError("Something went wrong"))
+        }
+
+        if(!passwordChecker.isValid) {
+            return UseCaseResult.ErrorPassword(passwordChecker.passwordError
+                ?: return UseCaseResult.GenericError("Something went wrong"))
+        }
+
+        val response = repository.signIn(email, password)
+
+        return when(response) {
+            is AuthResult.Authorized -> UseCaseResult.Authorized
```

Comment:
If you make a generic use case result, it should also be applicable for all use cases. If it contains a `UseCaseResult.Authorized`, it's not generally applicable, since that's clearly related to authentication and nothing else. 
You could simply create a class called `Result` inside the Login use case here which contains this, so each use case that needs a result has its own one and accessing it via `Login.Result` makes it clear what kind of result it is.